src/FileSystem-Path/AbsolutePath.class.st

```diff
@@ -27,7 +27,11 @@ AbsolutePath >> isAbsolute [
 
 { #category : #testing }
 AbsolutePath >> isRoot [
-	^ self size = 0
+	^self size = 0 or: [
+		JS
```

Comment:
That does not looks good. I think we should not have knowledge of JS environment in core classes.
Why "self size = 0" is not right condition for JS?

Reply:
If you want the system to come up on the Web VM, then there will be some decisions that must be delegated to the current host platform. To avoid mentioning any Web-related objects directly in the core classes, there should be additional obligatory protocol in the platform classes.

These changes are just meant to illustrate the codepoints where there must be additional collaboration with the host platform.

Reply:
That's fine @ccrraaiigg . But still why isRoot needs modification? Do you have concrete example what was not working considering aPath? 
We should refactor FileSystem to address the found problem. So that WebFileSystem will be the only required part

Reply:
Agreed. As for the root name size: at the moment, the filesystem that the SqueakJS VM creates from IndexedDB has a root hardwired with the name "/SqueakJS" instead of just "/". This is changeable.